api/v1beta1/common_types.go

```diff
@@ -45,18 +45,18 @@ const (
 	// When removing also remove from staticcheck exclude-rules for SA1019 in golangci.yml.
 	ClusterTopologyManagedFieldsAnnotation = "topology.cluster.x-k8s.io/managed-field-paths"
 
-	// ClusterTopologyMachineDeploymentLabelName is the label set on the generated  MachineDeployment objects
+	// ClusterTopologyMachineDeploymentNameLabel is the label set on the generated  MachineDeployment objects
 	// to track the name of the MachineDeployment topology it represents.
-	ClusterTopologyMachineDeploymentLabelName = "topology.cluster.x-k8s.io/deployment-name"
+	ClusterTopologyMachineDeploymentNameLabel = "topology.cluster.x-k8s.io/deployment-name"
 
 	// ClusterTopologyUnsafeUpdateClassNameAnnotation can be used to disable the webhook check on
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
-	// ProviderLabelName is the label set on components in the provider manifest.
+	// ProviderLabel is the label set on components in the provider manifest.
 	// This label allows to easily identify all the components belonging to a provider; the clusterctl
 	// tool uses this label for implementing provider's lifecycle operations.
-	ProviderLabelName = "cluster.x-k8s.io/provider"
+	ProviderLabel = "cluster.x-k8s.io/provider"
```

Comment:
q: should this be 
```suggestion
	ProviderNameLabel = "cluster.x-k8s.io/provider"
```
@sbueringer opinions?

Reply:
I would go with
>I would suggest to only include a Name in the constant name if the label actually contains name. In that case we should include it before Label and not after.
> https://github.com/kubernetes-sigs/cluster-api/issues/7611

If the label itself is wrong and should be `cluster.x-k8s.io/provider-name` that is something that should be changed independently.

I would keep the mapping from label string to constant name simple and if our label strings are wrong let's fix them if we think it's worth the effort.